gnd/src/main/java/com/google/android/gnd/persistence/sync/PhotoSyncWorker.java

```diff
@@ -66,7 +72,8 @@ public Result doWork() {
       try {
         remoteStorageManager
             .uploadMediaFromFile(new File(localSourcePath), remoteDestinationPath)
-            .blockingAwait();
+            .blockingForEach(
```

Comment:
Are we showing a notification for each photo being uploaded? Ideally we'd have a single status bar for sync progress.

Reply:
For now, yes. I plan to add notifications to all workers, then move on to refactor PhotoSyncWorker (merge with DataSyncWorker and store mutations in db)

Reply:
I may be wrong, but isn't this fundamentally different than showing a single always-on service UI notification? Won't we be doing a lot of work that will then need to be deleted?

Reply:
Nope, it's just a matter of changing boolean values to keep it sticky and then reuse the same notification for displaying updates. My idea was to have many eyes look at the code without making too many changes at once. But if you think it's better to do all at once, I can give it a try.

Reply:
If it's an incremental change then SGTM!